plugins/sfc/renderer/l2xconn/l2xconn_renderer.go

```diff
@@ -187,3 +217,126 @@ func (rndr *Renderer) getSFInterface(sf *renderer.ServiceFunction, input bool) s
 	}
 	return ""
 }
+
+//*******************************************
+// Functions added to manage multi-node SFCs
+//*******************************************
+
+// Return the node ID for the given SF
+func (rndr *Renderer) getSFNodeID(sf *renderer.ServiceFunction) uint32 {
+	switch sf.Type {
+	case renderer.Pod:
+		if len(sf.Pods) == 0 {
+			return 0
+		}
+		pod := sf.Pods[0] // TODO: handle chains with multiple pod instances per service function?
+		return pod.NodeID
+
+	case renderer.ExternalInterface: // TODO : check if the returned  nodeID  is correct (the required one) for ExternalInterface
+		// return nodeID of the first local interface
+		for _, extIf := range sf.ExternalInterfaces {
+			if extIf.Local {
+				return extIf.NodeID
+			}
+		}
+		// TODO: chain to a remote external interface?
+		return 0
+	}
+	return 0
+
+}
+
+// Check weather the given sf is in the local node or not
+func (rndr *Renderer) checkSFNodeLocal(sf *renderer.ServiceFunction) bool {
+	if sf.Type != renderer.Pod {
+		return false // TODO: to be checked if is it possible:  no render for the SFC
```

Comment:
It is possible - Interface SF has the `local` flag:

https://github.com/contiv/vpp/blob/dev/plugins/sfc/renderer/api.go#L165-L166

Reply:
The SF could be  a pod or an ExternalInterfaces. For Pod we have the local flag but for ExternalInterfaces we have to check if there is a local ExternalInterface or not --> I have changed the code of this Function in the new PR.